libtransmission/torrent-files.cc

```diff
@@ -243,17 +244,52 @@ void tr_torrent_files::remove(std::string_view parent_in, std::string_view tmpdi
         return;
     }
 
-    // make a tmpdir
+    // try to make a tmpdir
     auto tmpdir = tr_pathbuf{ parent, '/', tmpdir_prefix, "__XXXXXX"sv };
-    tr_sys_dir_create_temp(std::data(tmpdir));
+    if (!tr_sys_dir_create_temp(std::data(tmpdir), error))
+    {
+        return;
+    }
 
     // move the local data to the tmpdir
-    auto const paths = std::array<std::string_view, 1>{ parent.sv() };
-    for (tr_file_index_t idx = 0, n_files = fileCount(); idx < n_files; ++idx)
     {
-        if (auto const found = find(idx, std::data(paths), std::size(paths)); found)
+        struct moved_file
         {
-            tr_file_move(found->filename(), tr_pathbuf{ tmpdir, '/', found->subpath() });
+            tr_pathbuf from;
+            tr_pathbuf to;
+        };
+        std::vector<moved_file> moved_files;
+        bool successfully_moved = true;
+
+        auto const paths = std::array<std::string_view, 1>{ parent.sv() };
+        for (tr_file_index_t idx = 0, n_files = fileCount(); idx < n_files; ++idx)
+        {
+            if (auto const found = find(idx, std::data(paths), std::size(paths)); found)
+            {
+                moved_file f{ found->filename(), tr_pathbuf{ tmpdir, '/', found->subpath() } };
+
+                // if moving the file fails, stop
+                if (!tr_file_move_strict(f.from, f.to, error))
+                {
+                    successfully_moved = false;
+                    break;
+                }
+
+                moved_files.push_back(f);
+            }
+        }
+
+        // if moving failed for some file, rollback
```

Comment:
Hum, that's over trying for a `remove` method. The `tmp` folder should be on the same disk as the files, so there wouldn't be disk space issues. And the `tmp` folder was created by Transmission, so there wouldn't be permission issues at the destination  (only at the source).

So my feeling is that we don't have to bother rolling back:
1. We remove the `break` from line 275 to move as many files as we can to that tmp
2. We remove that tmp folder

And job is done to our best efforts: nothing else to attempt.

Reply:
The issue I had was precisely because some error was ignored when the disk was full, and I am not sure I trust all moves to work on a full disk (e.g. a directory might need to allocate a new page), so I don't like the idea of silently ignoring the error.

Maybe no rollback attempt should be made, but instead we should just raise an error?

Reply:
I don't mind the error. But no rollback please.

Reply:
I agree that the rollback is too much.

Reply:
Fixed.